src/main/java/store/domain/customer/Customer.java

```diff
@@ -0,0 +1,121 @@
+package store.domain.customer;
+
+import java.util.List;
+
+import store.domain.notice.CantPromotionNotice;
+import store.domain.notice.FreePromotionNotice;
+import store.domain.notice.Notice;
+import store.domain.notice.NoticeType;
+import store.domain.store.Promotion;
+import store.domain.store.Stock;
+import store.exception.state.InvalidNoticeTypeException;
+
+public class Customer {
+
+    private static final int NO_BONUS = 0;
+
+    private final Orders orders = new Orders();
+    private final Notices notices = new Notices();
+    private Membership membership = new Membership(false);
+
+    public void notice(Notice notice) {
+        notices.add(notice);
+    }
+
+    public void order(Stock stock, int quantity, int bonusQuantity, boolean onPromotion) {
+        if (quantity > 0) {
+            orders.add(new Order(stock, quantity, bonusQuantity, onPromotion));
+            stock.buy(quantity);
+        }
+    }
```

Comment:
음 이건 제 개인적인 생각입니다만,, order() 메서드가 여러곳에서 호출되고 있다보니 코드가 한눈에 들어오지 않는 것 같아요,,!
아래와 같이 개선하면 역할이 더욱 명확히 보이지 않을까요?? 선권님의 의견은 어떠신가요?
```suggestion
    public void orderWithBonus(Stock stock, int quantity, int bonusQuantity) {
        if (quantity > 0) {
            processOrder(stock, quantity, bonusQuantity, true);
        }
    }

    public void orderWithoutBonus(Stock stock, int quantity) {
        if (quantity > 0) {
            processOrder(stock, quantity, NO_BONUS, false);
        }
    }

    private void processOrder(Stock stock, int quantity, int bonusQuantity, boolean onPromotion) {
        orders.add(new Order(stock, quantity, bonusQuantity, onPromotion));
        stock.buy(quantity);
    }
```

Reply:
오... 정말 좋은 방법인 것 같아요!! 💯  완전 동의합니다!
좋은 의견 감사합니다 😄